.github/workflows/docs_environment.yml

```diff
@@ -0,0 +1,26 @@
+name: pygmt
```

Comment:
inspiration from xarray - https://github.com/pydata/xarray/tree/main/ci/requirements

Reply:
Sounds a good place to put all CI related files. 

Ping @weiji14 for comments on this.

Reply:
Ideally we would put all these 'docs' dependencies as an 'extras' in pyproject.toml/setup.py following https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies so that we could just `pip install .[docs]`. However, GMT would still need to come from conda-forge, unless #1853 becomes a thing :slightly_smiling_face:

Reply:
Actually, is the GMT binary needed to build the docs? Could we just mock it out? I recall @meghanrjones was able to use https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_mock_imports at https://github.com/pangeo-data/xbatcher/pull/62#issuecomment-1117833077, though not sure if it works in this case.

Reply:
> Actually, is the GMT C library needed to build the docs? Could we just mock it out?

I don't think it's possible. We need to call the GMT C library to generate images (e.g., all the images generated by sphinx-gallery).